lxd/api_project.go

```diff
@@ -764,14 +765,17 @@ func isEitherAllowOrBlockOrManaged(value string) error {
 func projectValidateConfig(s *state.State, config map[string]string) error {
 	// Validate the project configuration.
 	projectConfigKeys := map[string]func(value string) error{
-		"backups.compression_algorithm":  validate.IsCompressionAlgorithm,
-		"features.profiles":              validate.Optional(validate.IsBool),
-		"features.images":                validate.Optional(validate.IsBool),
-		"features.storage.volumes":       validate.Optional(validate.IsBool),
-		"features.networks":              validate.Optional(validate.IsBool),
-		"images.auto_update_cached":      validate.Optional(validate.IsBool),
-		"images.auto_update_interval":    validate.Optional(validate.IsInt64),
-		"images.compression_algorithm":   validate.IsCompressionAlgorithm,
+		"backups.compression_algorithm": validate.IsCompressionAlgorithm,
+		"features.profiles":             validate.Optional(validate.IsBool),
+		"features.images":               validate.Optional(validate.IsBool),
+		"features.storage.volumes":      validate.Optional(validate.IsBool),
+		"features.networks":             validate.Optional(validate.IsBool),
+		"images.auto_update_cached":     validate.Optional(validate.IsBool),
+		"images.auto_update_interval":   validate.Optional(validate.IsInt64),
+		"images.compression_algorithm":  validate.IsCompressionAlgorithm,
+		"images.default_architecture": func(value string) error {
+			return validate.IsOneOf(value, osarch.SupportedArchitectures())
+		},
```

Comment:
Could probably define a `validate.IsArchitecture` function to make this even easier to reuse elsewhere.

Reply:
@stgraber Yes, this is how I wanted to do this but after import `osarch` package in `validate.go` I get error related with cycle import:
```
import cycle not allowed
package github.com/lxc/lxd/client
	imports github.com/lxc/lxd/shared
	imports github.com/lxc/lxd/shared/validate
	imports github.com/lxc/lxd/shared/osarch
	imports github.com/lxc/lxd/shared
```

so this is why I made this in this way.

Reply:
Ah, that's slightly annoying :)

Reply:
Fixed that for you now.